pkg/clusterd/disk.go

```diff
@@ -66,6 +66,24 @@ func DiscoverDevices(executor exec.Executor) ([]*sys.LocalDisk, error) {
 			// go on without udev info
 			logger.Warningf("failed to get udev info for device %s: %+v", d, err)
 		}
+
+		// Test if device has child, if so we skip it and only consider the partitions
+		// which will come in later iterations of the loop
+		// We only test if the type is 'disk', this is a property reported by lsblk
+		// and means it's a parent block device
+		if disk.Type == sys.DiskType {
```

Comment:
Where does the partition get added to the list of available devices? This just looks like it skips the devices if they have partitions.

Reply:
The partition appears from `sys.ListDevices(executor)`.
The property is added by `PopulateDeviceInfo()`.
Yes, if the device has partitions we don't want to report it.

For instance, if you have:

* `/dev/vdb`
* `/dev/vdb1`
* `/dev/vdb2`

You don't want to pass `/dev/vdb` to c-v, but only the partitions (if they are good candidates).